router/internal/retrytransport/retry_transport.go

```diff
@@ -59,8 +59,8 @@ func NewRetryHTTPTransport(roundTripper http.RoundTripper, retryOptions RetryOpt
 func (rt *RetryHTTPTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 
 	resp, err := rt.RoundTripper.RoundTrip(req)
-	// Short circuit if the request was successful
-	if err == nil && resp.StatusCode == http.StatusOK {
+	// Short circuit if the request was successful.
+	if err == nil && isResponseOK(resp) {
```

Comment:
Do we want to retry 4xx responses? e.g. bad request, unauthorized, etc..? (I don't think so)

Reply:
Correct, only on network and retryable 4xx and 5xx codes. List: https://github.com/wundergraph/cosmo/blob/cbc103d2e2afe32c30d925d3ab04ddf72e614b99/router/internal/retrytransport/retry_transport.go#L26